### PR TITLE
[quant][fx] Add support for GRU in fx graph mode quantization

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91976

Summary:
might be needed by a meta-internal use case

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_rnn

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: torch/ao/nn/quantized/reference/modules/rnn.py [hunk 1]
```diff
@@ -488,3 +488,124 @@ def from_float(cls, mod, weight_qparams_dict):
...
+          ...
+        }
+        """
+        quantized_weight_bias_dict = {}
+        for wn in self._flat_weights_names:
+            if hasattr(self, wn):
+                if wn.startswith("weight"):
+                    weight_or_bias = get_quantized_weight(self, wn)
+                else:
+                    weight_or_bias = getattr(self, wn)
```
Comment (jcaip) [inline torch/ao/nn/quantized/reference/modules/rnn.py:518]: just curious - is it customary to not quantize bias at all?

File: torch/ao/nn/quantized/reference/modules/rnn.py [hunk 2]
```diff
@@ -488,3 +488,124 @@ def from_float(cls, mod, weight_qparams_dict):
...
+            hx = self.permute_hidden(hx, sorted_indices)
+
+        self.check_forward_args(input, hx, batch_sizes)
+        if batch_sizes is None:
+            result = _VF.gru(input, hx, self.get_flat_weights(), self.bias, self.num_layers,
+                             self.dropout, self.training, self.bidirectional, self.batch_first)
+        else:
+            result = _VF.gru(input, batch_sizes, hx, self.get_flat_weights(), self.bias,
+                             self.num_layers, self.dropout, self.training, self.bidirectional)
+        output = result[0]
```
Comment (jcaip) [inline torch/ao/nn/quantized/reference/modules/rnn.py:581]: nit: does `output, hidden = result` work here?

**Review by jcaip — APPROVED** [review]
lgtm

File: torch/ao/nn/quantized/reference/modules/rnn.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/nn/quantized/reference/modules/rnn.py:518]: it's a design choice when we are designing eager mode quantization. in computation, bias is actually quantized to int32 with the scale of input and weight, but we only have this information in graph mode quantization, not in eager, so we have to keep bias fp32 and quantize it internally in the quantized linear operator

File: torch/ao/nn/quantized/reference/modules/rnn.py (hunk 2 above)
Comment (jerryzh168) [reply torch/ao/nn/quantized/reference/modules/rnn.py:581]: probably, this is just copied from forward function of GRU though, will add a comment in the beginning

**Pull request closed without merging**